cst438Assignment2/src/main/java/cst438hw2/controller/CityRestController.java

```diff
@@ -0,0 +1,30 @@
+package cst438hw2.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import cst438hw2.domain.*;
+import cst438hw2.service.CityService;
+
+@RestController
+public class CityRestController
+{
+   @Autowired
+   private CityService cityService;
+   
+   @GetMapping("/api/cities/{city}")
+   public ResponseEntity<CityInfo> getWeather(@PathVariable("city") String cityName)
+   {
+      CityInfo cityInfo = cityService.getCityInfo(cityName);
+      
+      if (cityInfo != null) {
+         return new ResponseEntity<CityInfo>(cityInfo, HttpStatus.OK);
+      }
+      //Not sure what to pass as best practice
+      return new ResponseEntity<CityInfo>(HttpStatus.NO_CONTENT);
```

Comment:
I think for this you would just put NOT_FOUND instead of NO_CONTENT. NO_CONTENT would suggest that a city exists, but there is no data for the city vs just saying that the city doesn't exist with NOT_FOUND